### PR TITLE
Create project and feed tweaks

### PR DESCRIPTION
**Pull request by landonreed**
A few tweaks for #41:

- auto focus first field in form
- use `validator` library to validate URL (which is already used in editor)
- add keydown listener for `ENTER` (to trigger save)
- show loading spinner after save to prevent form from emptying and looking like something went wrong

### CODE REVIEW

File: lib/manager/components/CreateFeedSource.js [hunk 1]
```diff
@@ -36,13 +37,29 @@ type Validation = {
...
+  }
+
+  componentWillUnmount () {
+    window.removeEventListener('keydown', this._handleKeyDown)
+  }
+
+  _handleKeyDown = (e: SyntheticKeyboardEvent<HTMLInputElement>) => {
+    switch( e.keyCode ) {
+      case 13: // ENTER
+        this._onSave()
```
Comment (evansiroky) [inline lib/manager/components/CreateFeedSource.js:58]: There needs to be some kind of validation check done before calling `_onSave`.  It's possible to create a feed source without a name with this change.

File: lib/manager/components/ProjectSettingsForm.js [hunk 2]
```diff
@@ -78,6 +78,21 @@ export default class ProjectSettingsForm extends MessageComponent<Props, State>
...
+  }
+
+  componentWillUnmount () {
+    window.removeEventListener('keydown', this._handleKeyDown)
+  }
+
+  _handleKeyDown = (e: SyntheticKeyboardEvent<HTMLInputElement>) => {
+    switch( e.keyCode ) {
+      case 13: // ENTER
+        this._onSaveSettings()
```
Comment (evansiroky) [inline lib/manager/components/ProjectSettingsForm.js:91]: There needs to be some kind of validation check done before calling `_onSaveSettings`.  It's possible to create a project without a name with this change.

**Review by evansiroky — CHANGES_REQUESTED** [review]
There ought to be a validation check after pressing enter and before saving a feed source or project.  It'd probably be good to recalculate the form validation when enter is pressed and if everything looks good, continue with saving.

**Comment by landonreed** [comment]
OK, @evansiroky, I made some changes that address the onSave issue. Let me know how it looks.

**Review by evansiroky — APPROVED** [review]

**Pull request merged**